File: FxPricer/src/hu/fx/service/config/ConfigReader.scala
package hu.fx.service.config

import java.io.File
import java.io.FileInputStream
import java.util.Properties

object ConfigReader {

  private val CONFIG_PATH = "resources/props_default.properties"
  private val properties = ConfigReader.readProperties()
  
  def getProperty(key: String): Option[String] = {
    val value = properties.getProperty(key)
    if (value == null) None
    else Some(value)
  }

  private object ConfigReader {
    def readProperties() = {
      val props = new Properties
      val loaded = props.load(new FileInputStream(new File(CONFIG_PATH)))
      props
    }
  }
}

File: MonitoringService/src/main/java/hu/fx/config/ConfigReader.java
package hu.fx.config;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class ConfigReader {

	public Properties getProperties() {
		Properties properties = new Properties();
		InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream("WEB-INF/props_default.properties");
		
		try {
			properties.load(inputStream);
		} catch (IOException e) {
			throw new IllegalStateException(e);
		}
		
		return properties;
	}
}
